tests/helpers/test_reports_helpers.py

```diff
@@ -144,6 +160,170 @@ def test_get_cherrypicked_samples_repeat_tests(
         pd.testing.assert_frame_equal(expected, returned_samples)
 
 
+def test_get_cherrypicked_samples_no_sentinel(app, freezer):
+    expected = [
+        pd.DataFrame([]),  # Sentinel query response
+        pd.DataFrame(
+            ["MCM001", "MCM003", "MCM005"], columns=[FIELD_ROOT_SAMPLE_ID], index=[0, 1, 2]
+        ),  # Beckman query response
+    ]
+    samples = ["MCM001", "MCM002", "MCM003", "MCM004", "MCM005"]
+    plate_barcodes = ["123", "456"]
+
+    with app.app_context():
+        with patch("sqlalchemy.create_engine", return_value=Mock()):
+            with patch(
+                "pandas.read_sql",
+                side_effect=expected,
+            ):
+                returned_samples = get_cherrypicked_samples(samples, plate_barcodes)
+                assert returned_samples.at[0, FIELD_ROOT_SAMPLE_ID] == "MCM001"
+                assert returned_samples.at[1, FIELD_ROOT_SAMPLE_ID] == "MCM003"
+                assert returned_samples.at[2, FIELD_ROOT_SAMPLE_ID] == "MCM005"
+
+
+def test_get_cherrypicked_samples_chunking_no_sentinel(app, freezer):
+    # Note: This represents the results of three different (Sentinel, Beckman) sets of
+    # database queries, each Sentinel query getting indexed from 0. Do not changes the
+    # indicies here unless you have modified the behaviour of the query.
+    query_results = [
+        pd.DataFrame([]),
+        pd.DataFrame(["MCM001"], columns=[FIELD_ROOT_SAMPLE_ID], index=[0]),
+        pd.DataFrame([]),
+        pd.DataFrame(["MCM003"], columns=[FIELD_ROOT_SAMPLE_ID], index=[0]),
+        pd.DataFrame([]),
+        pd.DataFrame(["MCM005"], columns=[FIELD_ROOT_SAMPLE_ID], index=[0]),
+    ]
+    expected = pd.DataFrame(
+        ["MCM001", "MCM003", "MCM005"], columns=[FIELD_ROOT_SAMPLE_ID], index=[0, 1, 2]
+    )
+
+    samples = ["MCM001", "MCM002", "MCM003", "MCM004", "MCM005"]
+    plate_barcodes = ["123", "456"]
+
+    with app.app_context():
+        with patch("sqlalchemy.create_engine", return_value=Mock()):
+            with patch(
+                "pandas.read_sql",
+                side_effect=query_results,
+            ):
+                returned_samples = get_cherrypicked_samples(samples, plate_barcodes, 2)
+                pd.testing.assert_frame_equal(expected, returned_samples)
+
+
+# test scenario where there have been multiple lighthouse tests for a sample with the same Root
+# Sample ID uses actual databases rather than mocking to make sure the query is correct
+def test_get_cherrypicked_samples_repeat_tests_no_sentinel(
+    app, freezer, mlwh_beckman_cherrypicked, event_wh_data
+):
+    # the following come from MLWH_SAMPLE_LIGHTHOUSE_SAMPLE in fixture_data
+    root_sample_ids = ["root_4", "root_5", "root_4"]
+    plate_barcodes = ["pb_4", "pb_5", "pb_6"]
+
+    # root_4 will match 2 samples, but only one of those will match an event (on sample uuid)
```

Comment:
yep, good, i like the comments clearly describing why we get the result in your tests